app/farm/views.py

```diff
@@ -0,0 +1,42 @@
+from django.http import HttpResponse, HttpResponseRedirect
+from django.shortcuts import render, get_object_or_404
+from django.urls import reverse
+
+from .models import Farm
+
+
+def index(request):
+    # return HttpResponse("You're looking at a test farm")
+    farm_list = Farm.objects.order_by('-name')[:5]
+    context = {'farm_list': farm_list}
+    return render(request, 'farm/index.html', context)
+
+
+def details(request, farm_id):
+    farm = get_object_or_404(Farm, pk=farm_id)
+    return render(request, 'farm/details.html', {'farm': farm})
+
+
+def add(request):
+    farm = Farm(
+        name=request.POST['name'],
```

Comment:
When pressing the `Add New` button, the `add` route results in a `MultiValueDictKeyError` error, as it is trying to access the `name` in the POST part of request. Is this meant to return a form on the `GET` request?

Reply:
fixed